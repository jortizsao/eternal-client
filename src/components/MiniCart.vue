<template lang="html">
  <div class="minicart">
    <div v-if="cart && cart.lineItems > 0">
      <button class="not-empty link-your-bag icon-hand-bag">
        <span class="hidden-xs hidden-sm">Cart</span>
        <span class="cart-item-number">{{cart.lineItems.length}}</span>
      </button>
      <div class="col-xs-12">
        <div class="nav-minicart">
        </div>
      </div>
    </div>
    <div v-else>
      <a href="" class="not-empty link-your-bag icon-hand-bag">
        <span class="hidden-xs hidden-sm">Cart</span>
        <span class="cart-item-number">0</span>
      </a>
    </div>
  </div>

</template>

<script>
export default {
  data() {
    return {};
  },
  props: ['cart'],
};
</script>

<style lang="css">
</style>
