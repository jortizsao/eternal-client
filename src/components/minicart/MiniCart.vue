<template lang="html">
  <div class="minicart">
    <template v-if="cart && cart.lineItems.length">
      <button class="not-empty link-your-bag icon-hand-bag">
        <span class="hidden-xs hidden-sm cart">{{ $t('miniCart') }}</span>
        <span class="cart-item-number">{{cart.lineItems.length}}</span>
      </button>
      <div class="col-xs-12">
        <div class="nav-minicart">
        </div>
      </div>
    </template>
    <template v-else>
      <a href="" class="not-empty link-your-bag icon-hand-bag">
        <span class="hidden-xs hidden-sm cart">{{ $t('miniCart') }}</span>
        <span class="cart-item-number">0</span>
      </a>
    </template>
  </div>

</template>

<script>
export default {
  data() {
    return {};
  },
  props: ['cart'],
};
</script>

<i18n>
en:
  miniCart: Cart
de:
  miniCart: Warenkorb
</i18n>
