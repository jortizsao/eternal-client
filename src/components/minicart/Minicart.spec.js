import { shallow } from 'vue-test-utils';
import { createRenderer } from 'vue-server-renderer';
import MiniCart from './MiniCart.vue';

describe('Minicart', () => {
  describe(':props', () => {
    describe(':cart', () => {
      let renderer;
      let wrapper;

      beforeEach(() => {
        renderer = createRenderer();
        wrapper = shallow(MiniCart, {
          mocks: {
            $t: () => {},
          },
        });
      });

      afterEach(() => {
        renderer.renderToString(wrapper.vm, (err, str) => {
          expect(str).toMatchSnapshot();
        });
      });

      it('should render the number of items in the cart', () => {
        wrapper.setProps({
          cart: {
            lineItems: [
              {
                id: '1',
                name: 'product1',
              },
            ],
          },
        });
      });

      it('should render 0 items when there is no cart', () => {
        wrapper.setProps({});
      });
    });
  });
});
