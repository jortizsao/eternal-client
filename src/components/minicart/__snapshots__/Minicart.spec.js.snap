// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Minicart :props :cart should render 0 items when there is no cart 1`] = `
<div class="minicart">
    <a href="" class="not-empty link-your-bag icon-hand-bag"><span class="hidden-xs hidden-sm cart"></span> <span class="cart-item-number">0</span></a>
</div>
`;

exports[`Minicart :props :cart should render the number of items in the cart 1`] = `
<div class="minicart">
    <button class="not-empty link-your-bag icon-hand-bag"><span class="hidden-xs hidden-sm cart"></span> <span class="cart-item-number">1</span></button>
    <div class="col-xs-12">
        <div class="nav-minicart"></div>
    </div>
</div>
`;
